server/app/views/components/buttons/ButtonAction.java

```diff
@@ -0,0 +1,36 @@
+package views.components.buttons;
+
+import com.google.auto.value.AutoOneOf;
+
+/**
+ * Represents the action that should be taken when the end user clicks a button. As an AutoOneOf,
+ * exactly one option from the enum will be selected at a time.
+ */
+@AutoOneOf(ButtonAction.Action.class)
+public abstract class ButtonAction {
+  public enum Action {
+    REDIRECT,
+    SUBMIT_WITH_FORM_ID,
+    SUBMIT_WITHOUT_FORM_ID
+  }
+
+  public abstract Action action();
+
+  public static ButtonAction ofRedirect(String redirectUrl) {
```

Comment:
Is the "of" prefix standard? Feels more readable to me to omit it

Reply:
It's recommended by the how-to guide:
https://github.com/google/auto/blob/main/value/userguide/howto.md#-make-a-class-where-only-one-of-its-properties-is-ever-set

Open to changing if you think it's better.